client/landing/stepper/declarative-flow/internals/steps-repository/wait-for-atomic/index.tsx

```diff
@@ -32,13 +32,17 @@ export const transferStates = {
 
 const wait = ( ms: number ) => new Promise( ( res ) => setTimeout( res, ms ) );
 
-const WaitForAtomic: Step = function WaitForAtomic( { navigation } ) {
+const WaitForAtomic: Step = function WaitForAtomic( { navigation, data } ) {
 	const { submit } = navigation;
 	const { setPendingAction, setProgress } = useDispatch( ONBOARD_STORE );
 	const { requestLatestAtomicTransfer } = useDispatch( SITE_STORE );
 	const site = useSite();
 
-	const siteId = site?.ID;
+	let siteId = site?.ID as number;
+	// In some cases, we get the siteId from the navigation extra data rather than the SITE_STORE.
```

Comment:
Why (out of curiosity)?

Reply:
Because the WooExpress flow doesn't require any user interaction, it doesn't use the data stores in the same way as the other flows. So instead of a user initiating the next step and it populating the `SITE_STORE`, the WooExpress `siteCreationStep` passes the newly created site information on to the next step via the extra-data functionality we have.